modules/generic/src/main/scala/pureconfig/generic/EnumerationConfigReaderBuilder.scala

```diff
@@ -0,0 +1,52 @@
+package pureconfig.generic
+
+import pureconfig.ConfigReader.Result
+import pureconfig.error.NoValidCoproductChoiceFound
+import pureconfig.{ ConfigCursor, ConfigReader }
+import shapeless._
+import shapeless.labelled._
+
+trait EnumerationConfigReaderBuilder[A] {
```

Comment:
Can you add Scaladoc to every new type class and their respective methods?

Reply:
I added some Scaladoc in a26cebd.